### ISSUE TITLE
[Bug]: Segment with deleted data can not be loaded

### ISSUE THREAD

**Issue by xiaofan-luan** [milvus-io/milvus#16821]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

From the log, we are trying to do segmentLoadDeletedRecord on a growing segment but it should only happen on a sealed one, need to investigate why this is happening.

### Expected Behavior

load should successful with deleted data.

### Milvus Log

```markdown
[ERROR] [impl.go:238] ["segmentLoadFieldData failed, illegal segment type Growing segmentID = 432990615898947585\n"] [stack="github.com/milvus-io/milvus/internal/querynode.(*QueryNode).WatchDmChannels.func1\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:238\ngithub.com/milvus-io/milvus/internal/querynode.(*QueryNode).WatchDmChannels\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:247\ngithub.com/milvus-io/milvus/internal/distributed/querynode.(*Server).WatchDmChannels\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:269\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_WatchDmChannels_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:3504\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/querypb._QueryNode_WatchDmChannels_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:3506\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"
[ERROR] [segment_loader.go:138] ["load segment failed when load data into memory"] [collectionID=432989626059194369] [partitionID=432989626059194370] [segmentID=432990615898947585] ["segment type"=2] [error="segmentLoadFieldData failed, illegal segment type Growing segmentID = 432990615898947585\n"] [stack="github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadSegment\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:138\ngithub.com/milvus-io/milvus/internal/querynode.(*watchDmChannelsTask).Execute\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task.go:311\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:61\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).taskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:78"]
```

**Comment by congqixia** [comment]
Growing Segment with valid checkpoint may have delta log. Maybe the restriction shall be removed. @xiaofan-luan @soothing-rain

**Comment by soothing-rain** [comment]
Fixed in 2.0.2 hotfix and in latest master.

**Comment by xiaofan-luan** [comment]
close it for now. rest of the loading issue is assigned to @xige-16
